tests/Doctrine/Tests/ORM/Functional/Ticket/GH6823Test.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\DBAL\Platforms\MySQLPlatform;
+use Doctrine\ORM\Mapping\Column;
+use Doctrine\ORM\Mapping\Entity;
+use Doctrine\ORM\Mapping\Id;
+use Doctrine\ORM\Mapping\ManyToOne;
+use Doctrine\ORM\Mapping\Table;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+class GH6823Test extends OrmFunctionalTestCase
+{
+    public function testCharsetCollationWhenCreatingForeignRelations(): void
+    {
+        if (! $this->_em->getConnection()->getDatabasePlatform() instanceof MySQLPlatform) {
+            self::markTestSkipped('This test is useful for all databases, but designed only for mysql.');
+        }
+
+        $sql = $this->_schemaTool->getCreateSchemaSql(
+            [
+                $this->_em->getClassMetadata(GH6823User::class),
+                $this->_em->getClassMetadata(GH6823Group::class),
+                $this->_em->getClassMetadata(GH6823Status::class),
+            ]
+        );
```

Comment:
Please use `createSchemaForModels` for this.

Reply:
Done